src/main/java/com/celements/cleverreach/util/SynthonCssInliner.java

```diff
@@ -0,0 +1,160 @@
+package com.celements.cleverreach.util;
+
+import static com.google.common.base.Preconditions.*;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.io.OutputStream;
+import java.net.HttpURLConnection;
+import java.net.MalformedURLException;
+import java.net.ProtocolException;
+import java.net.URL;
+import java.net.URLEncoder;
+import java.nio.charset.StandardCharsets;
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+
+import javax.validation.constraints.NotNull;
+
+import org.codehaus.jackson.map.ObjectMapper;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.xwiki.component.annotation.Component;
+
+import com.celements.cleverreach.exception.CssInlineException;
+import com.google.common.base.Strings;
+import com.google.common.hash.Hashing;
+
+@Component("synthon")
+public class SynthonCssInliner implements CssInliner {
+
+  private static final Logger LOGGER = LoggerFactory.getLogger(SynthonCssInliner.class);
+
+  public static final String CONFIG_KEY_INLINE_URL = "inlineUrl";
+  public static final String CONFIG_KEY_SERVER_SECRET = "serverSecret";
+  public static final String CONFIG_KEY_REMOVE_CLASSES = "removeClasses";
+
+  @Override
+  public String inline(String html, List<String> cssList) throws CssInlineException {
+    return inline(html, cssList, null);
+  }
+
+  @Override
+  public @NotNull String inline(@NotNull String html, @NotNull List<String> cssList,
+      Map<String, String> configs) throws CssInlineException {
+    return inline(html, String.join("\n", cssList), configs);
+  }
+
+  @Override
+  public @NotNull String inline(@NotNull String html, @NotNull String css)
+      throws CssInlineException {
+    return inline(html, css, null);
+  }
+
+  /**
+   * Supported (optional) parameters:
+   * removeClasses [true|false] - remove the classes from the HTML after inlining
+   * inlineUrl - can be used to override the default URL used for inline service
+   */
+  @Override
+  public String inline(String html, String css, Map<String, String> configs)
+      throws CssInlineException {
+    checkNotNull(html);
+    checkNotNull(css);
+    configs = (configs == null) ? Collections.emptyMap() : configs;
+    LOGGER.trace("Applying the following CSS [{}] to HTML [{}]", css, html);
+    try {
+      String encodedHtml = merge(html, css);
+      if (!configs.containsKey(CONFIG_KEY_SERVER_SECRET)) {
+        throw new CssInlineException(encodedHtml, new RuntimeException("Server secret for synthon "
+            + "serveris missing in config"));
```

Comment:
add constructor `CssInlineException(String html, String msg)`
```suggestion
        throw new CssInlineException(encodedHtml, "Server secret for synthon serveris missing in config");
```